src/api/entities/CustomPermissionGroup.ts

```diff
@@ -0,0 +1,94 @@
+import BigNumber from 'bignumber.js';
+
+import { Context, PermissionGroup } from '~/internal';
+import { GroupPermissions } from '~/types';
+import {
+  extrinsicPermissionsToTransactionPermissions,
+  numberToU32,
+  stringToTicker,
+  transactionPermissionsToTxGroups,
+} from '~/utils/conversion';
+
+export interface HumanReadable {
+  id: BigNumber;
+  ticker: string;
+}
+
+export interface UniqueIdentifiers {
+  id: BigNumber;
+  ticker: string;
+}
+
+/**
+ * Represents a group of custom permissions for a Security Token
+ */
+export class CustomPermissionGroup extends PermissionGroup {
+  /**
+   * @hidden
+   * Check if a value is of type [[UniqueIdentifiers]]
+   */
+  public static isUniqueIdentifiers(identifier: unknown): identifier is UniqueIdentifiers {
+    const { id, ticker } = identifier as UniqueIdentifiers;
+
+    return id instanceof BigNumber && typeof ticker === 'string';
+  }
+
+  public id: BigNumber;
+  public ticker: string;
+
+  /**
+   * @hidden
+   */
+  public constructor(identifiers: UniqueIdentifiers, context: Context) {
+    super(identifiers, context);
+
+    const { id, ticker } = identifiers;
+
+    this.id = id;
+    this.ticker = ticker;
+  }
+
+  /**
+   * Return the Group's ID
+   */
+  public toJson(): HumanReadable {
+    const { id, ticker } = this;
+    return {
+      id,
+      ticker,
+    };
+  }
+
+  /**
+   * Retrieve the list of permissions and transaction groups associated with this Permission Group
+   */
+  public async getPermissions(): Promise<GroupPermissions> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { externalAgents },
+        },
+      },
+      context,
+      ticker,
+      id,
+    } = this;
+
+    const rawTicker = stringToTicker(ticker, context);
+    const rawAgId = numberToU32(id, context);
+
+    const rawGroupPermissions = await externalAgents.groupPermissions(rawTicker, rawAgId);
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const transactionPermissions = extrinsicPermissionsToTransactionPermissions(
+      rawGroupPermissions.unwrap()
+    )!;
```

Comment:
groupPermissions
`Promise<Option<ExtrinsicPermissions>>`

Reply:
transactionPermissionsToTxGroups expects TransactionPermissions, but in this case could be TransactionPermissions | null.
I assume every PG should have at least one transaction or transactionGroup associated to it

Reply:
I think the assertion is not a requirement. Given that `transactions` in `GroupPermissions` can be `TransactionPermissions | null` you should allow `extrinsicPermissionsToTransactionPermissions` to receive null too to avoid this assertion.

Reply:
I think the assertion is not a requirement. Given that `transactions` in `GroupPermissions` can be `TransactionPermissions | null` you should allow `extrinsicPermissionsToTransactionPermissions` to receive null too to avoid this assertion.

Reply:
I think the assertion is not a requirement. Given that `transactions` in `GroupPermissions` can be `TransactionPermissions | null` you should allow `extrinsicPermissionsToTransactionPermissions` to receive null too to avoid this assertion.

Reply:
the problem here is transactionPermissionsToTxGroups not accept null as argument, but extrinsicPermissionsToTransactionPermissions could returns TransactionPermissions or null

Reply:
`transactionPermissionsToTxGroups` should accept null as argument, returning all TxGroups. Again, `null` means `Whole` transactions for us and that's what we should be returning.

Reply:
We're using an empty TxGroup array when transaction permissions are Whole (I don't remember why we decided that, but that's the way it is right now and changing it would be a breaking change)

> the problem here is transactionPermissionsToTxGroups not accept null as argument, but extrinsicPermissionsToTransactionPermissions could returns TransactionPermissions or null

The assertion doesn't prevent the value from being null, it just tells the compiler that you don't expect it to be. The function can (and will) still return null even with the assertion and then you'll get an error when it does and you pass it to `transactionPermissionsToTxGroups`. The correct way of handling this is handling the case where it is null

Reply:
so if it is null, I should return []